### PR TITLE
[docs] minikube fabric documentation

### PR DESCRIPTION
**Pull request by Mohitv**
**Changelog**
- Add Minikube documentation
- Fix to deploy HF fabric on minikube

 

**Reviewed by**
@jagpreetsinghsasan 

 

**Linked issue**
#1796

### CODE REVIEW

File: docs/source/developer/bevel_minikube_setup.md [hunk 1]
```diff
@@ -9,6 +9,18 @@
 
 Before proceeding, first make sure that you've completed [Developer Pre-requisites](./dev_prereq.md).
 
+## Setup Minikube
+
+1. Install Minikube https://minikube.sigs.k8s.io/docs/start/ on VM
+
+2. Start minikube. The setup has been tested on a Ubuntu 20.4 VM which has a public ip
```
Comment (jagpreetsinghsasan) [inline docs/source/developer/bevel_minikube_setup.md:16]: Typo `20.04 VM`

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
There are 2 additional review comments from my side
1. The entire README.md isn't properly sectioned and navigating through it is not convenient.
2. Can you add one more troubleshooting point which says how to delete the minikube and the proxy completely from the system (in-case someone wants to start fresh again)?

File: docs/source/developer/bevel_minikube_setup.md (hunk 1 above)
Comment (Mohitv) [reply docs/source/developer/bevel_minikube_setup.md]: I have created new file for fabric deployments on minikube. kindly review

File: docs/source/developer/bevel_fabric_minikube_setup.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+   ```
+
+   ***   
+   **NOTE**: If you ever delete and recreate minikube, the above steps have to be repeated.
+   ***
+
+## Setup Hashicorp vault
+
+   1. There are two options to setup vault:
+   Setup hashicorp vault and run in dev mode. Install the vault following instructions at following [url](
```
Comment (jagpreetsinghsasan) [inline docs/source/developer/bevel_fabric_minikube_setup.md:64]: You can rename the second line here as `Install the vault using Hashicorp Vault [official documentation](link_to_their_docs)`

File: docs/source/developer/bevel_fabric_minikube_setup.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+   1. There are two options to setup vault:
+   Setup hashicorp vault and run in dev mode. Install the vault following instructions at following [url]( 
+   https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-install)
+   
+      ```bash
+      vault server -dev -dev-root-token-id root -dev-listen-address 0.0.0.0:8200
+      ```
+   OR
+
+   2. One can also use helm charts for vault and do the operations as mentioned in the Workshop series [here](
```
Comment (jagpreetsinghsasan) [inline docs/source/developer/bevel_fabric_minikube_setup.md:72]: Can you pinpoint the time around which one should start looking for the vault specifics?

File: docs/source/developer/bevel_fabric_minikube_setup.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+   ```bash
+   cp ~/.kube/config build/
+   ```   
+4. Open the above config file in build directory and update file path for certificate-authority, client-certificate and client-key  to point to the respective files copied in build directory.
+
+   ![](./../_static/minikube-config.jpg)
+
+   Update config `server` value to following:
+   ```bash
+   server: https://<specify public ip of VM>:18443
```
Comment (jagpreetsinghsasan) [inline docs/source/developer/bevel_fabric_minikube_setup.md:54]: Here, can you mention what VM are we talking about, and how can one get its public IP?

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]

**Comment by jagpreetsinghsasan** [comment]
One last review comment at this step, is to squash merge commits into 1

**Comment by suvajit-sarkar** [comment]
Kindly rebase it on the develop branch

**Comment by Mohitv** [comment]
Created a new PR https://github.com/hyperledger/bevel/pull/2164, to merge to develop branch and ensure single file is available for review

**Pull request closed without merging**